jaeger-core/src/main/java/com/uber/jaeger/samplers/ProbabilisticSampler.java

```diff
@@ -33,25 +33,38 @@
 public class ProbabilisticSampler implements Sampler {
   public static final String TYPE = "probabilistic";
 
-  private final long positiveSamplingBoundary;
-  private final long negativeSamplingBoundary;
-  private final double samplingRate;
-  private final Map<String, Object> tags;
+  private final double DELTA = 0.00001;
+  private long positiveSamplingBoundary;
+  private long negativeSamplingBoundary;
+  private double samplingRate;
+  private Map<String, Object> tags;
 
   public ProbabilisticSampler(double samplingRate) {
+    init(samplingRate);
+  }
+
+  public synchronized void update(double samplingRate) {
+    if(Math.abs(this.samplingRate - samplingRate) < DELTA) {
+      return;
+    }
+
+    init(samplingRate);
+  }
+
+  private void init(double samplingRate) {
     if (samplingRate < 0.0 || samplingRate > 1.0) {
-      throw new IllegalArgumentException(
-          "The sampling rate must be greater than 0.0 and less than 1.0");
+      throw new IllegalArgumentException("The sampling rate must be greater than 0.0 and less than 1.0");
     }
 
     this.samplingRate = samplingRate;
-    this.positiveSamplingBoundary = (long) (((1L << 63) - 1) * samplingRate);
-    this.negativeSamplingBoundary = (long) ((1L << 63) * samplingRate);
 
-    Map<String, Object> tags = new HashMap<>();
+    HashMap<String, Object> tags = new HashMap<>();
     tags.put(Constants.SAMPLER_TYPE_TAG_KEY, TYPE);
     tags.put(Constants.SAMPLER_PARAM_TAG_KEY, samplingRate);
     this.tags = Collections.unmodifiableMap(tags);
+
+    positiveSamplingBoundary = (long) (((1L << 63) - 1) * samplingRate);
```

Comment:
I agree with @black-adder - this increases the size of the diff for no reason. And I also prefer constructor to be consistent in its use of `this.` prefix - if it's used for something, better be used for everything.